scripts/validation.js

```diff
@@ -0,0 +1,58 @@
+import { form } from "./form_script.js";
+
+function setRequiredLabels() {
+  const labels = document.querySelectorAll("label");
+
+  for (const label of labels) {
+    const labelFor = label.for;
+    if (labelFor) {
+      const field = document.querySelector(`#${labelFor}`);
+      if (field.hasAttribute("required") && field.type != "radio") {
+        label.classList.add("required");
+      }
+    }
+  }
+}
+
+function applyValidation(inputs) {
+  inputs.forEach((input) =>
+    input.addEventListener("input", function (e) {
```

Comment:
I'd recommend to use event delegation instead of applying the events for each input